### PR TITLE
fix: org name validation

### PR DESCRIPTION
**Pull request by Kirpal**
## Description
When trying to use the new `--org-name` support I found that underscores were not being supported. After looking a little closer, the regex for the org name was incorrect by not enforcing that segments must start with a letter and by enforcing a min/max of three segments.

I updated the org name regex, invalid org error message, and added more tests for edge cases.

#### Sources:
- [Android rules](https://developer.android.com/studio/build/application-id)
- [iOS rules](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundleidentifier#discussion)
- [Flutter android validation](https://github.com/flutter/flutter/blob/76d5e62f42dffe3aba42bba9364c4469af1ff332/packages/flutter_tools/lib/src/commands/create_base.dart#L481)

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Review by felangel — CHANGES_REQUESTED** [review]
Before this can be merged we will need to update the template to support dynamic length for org name parts.

File: lib/src/commands/create.dart [hunk 1]
```diff
@@ -151,8 +150,10 @@ class CreateCommand extends Command<int> {
     if (!isValidOrgName) {
       throw UsageException(
         '"$name" is not a valid org name.\n\n'
-        'A valid org name has 3 parts separated by "."'
-        'and only includes alphanumeric characters and underscores'
+        'A valid org name has:\n'
+        '  - At least 2 parts separated by "."\n'
+        '  - Each part must start with a letter\n'
+        '  - Only includes alphanumeric characters and underscores'
```
Comment (felangel) [inline lib/src/commands/create.dart:156]: According to the iOS spec:

> only alphanumeric characters (A-Z, a-z, and 0-9), hyphens (-), and periods (.)

I don't think we should allow underscores.

Comment (Kirpal) [reply lib/src/commands/create.dart:156]: That's a great idea, I'll update this

**Review by felangel — APPROVED** [review]
